packages/graphql/src/translate/batch-create/unwind-create-visitors/UnwindCreateVisitor.ts

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Context } from "../../../types";
+import type Relationship from "../../../classes/Relationship";
+import type { Visitor, ICreateAST, INestedCreateAST } from "../GraphQLInputAST/GraphQLInputAST";
+import { getAutoGeneratedFields, fieldToSetParam } from "../utils";
+import Cypher from "@neo4j/cypher-builder";
+import createRelationshipValidationString from "../../create-relationship-validation-string";
+import type { CallbackBucket } from "../../../classes/CallbackBucket";
+import { filterTruthy } from "../../../utils/utils";
+import { Neo4jGraphQLError } from "../../../classes";
+
+type UnwindCreateScopeDefinition = {
+    unwindVar: Cypher.Variable;
+    parentVar: Cypher.Variable;
+    clause?: Cypher.Clause;
+}
+type GraphQLInputASTNodeRef = string;
+type UnwindCreateEnvironment = Record<GraphQLInputASTNodeRef, UnwindCreateScopeDefinition>
+
+export class UnwindCreateVisitor implements Visitor {
+    callbackBucket: CallbackBucket;
+    context: Context;
+    rootNode: Cypher.Node | undefined;
+    clause: Cypher.Clause | undefined;
+    environment: UnwindCreateEnvironment;
+
+    constructor(
+        unwindVar: Cypher.Variable,
+        callbackBucket: CallbackBucket,
+        context: Context,
+    ) {
+        this.callbackBucket = callbackBucket;
+        this.context = context;
+        this.environment = { root: { unwindVar: unwindVar, parentVar: new Cypher.Variable() } };
```

Comment:
mmm, I see this is a mutable property of the class, despite it only being used in the visit method, which is called only once.

Maybe we could, instead, save the "root" variables as properties, and use instead a variable inside the method "visitCreate"? This will reduce the scope of this variable